src/common_util/univariatesumcheck_test.py

```diff
@@ -0,0 +1,141 @@
+import random, unittest
+import numpy as np
+from src.common_util.poly import (
+    Polynomial, 
+    Basis, 
+    Scalar, 
+    construct_non_multiplicativegroup_vanishing_poly
+)
+from src.common_util.curve import left_rotate_root_of_unity
+from src.common_util.univariatesumcheck import (
+    create_col_mapping,
+    construct_H_I,
+    construct_Dx,
+    construct_rho_col_j, 
+    prove as univariate_sumcheck_prove
+)
+from src.common_util.lagrange import lagrange_basis
+from src.cq.setup import Setup # TODO: refactor to common_util
+
+
+""" 
+(i) e (t, [D]2) − e ([1]1u2, [1]2) = e ([R]1, [x]2) + e ([Q2]1, [zI ]2) 
+"""
+
+# lookup matrix
+M = [
+    [0, 0, 1, 0],
+    [0, 1, 0, 0],
+    [0, 0, 1, 0],
+]
+M2 = [
+    [0, 0, 1, 0],
+    [0, 0, 1, 0],
+    [0, 0, 1, 0],
+]
+# the table we look up on
+c = [1, 2, 3, 4]
+# the result be looked up
+a = [3, 2, 3]
+a2 = [3, 3, 3]
+# unique of a
+t = [2, 3]
+
+class UnivariateSumcheckTest(unittest.TestCase):
+    def test_matrix_encode(self) -> None:
+        col_mapping = create_col_mapping(M)
+        assert list(col_mapping) == [3, 2, 3]
+    
+    def test_construct_Dx(self) -> None:        
+        m = len(M) # number of rows = number of mu
+        N = len(c) # table size
+
+        V = Scalar.roots_of_unity(m) # set for mu
+        H = Scalar.roots_of_unity(N) # set for rho
+        H = left_rotate_root_of_unity(H)
```

Comment:
Do not have to rotate, you can just use 1 as the first item of SRS: 1, w, w^2...